packages/compass-sidebar/src/stores/store.js

```diff
@@ -24,16 +28,18 @@ store.onActivated = (appRegistry) => {
     store.dispatch(changeConnection(legacyConnectionModel));
   });
 
-  appRegistry.on('instance-refreshed', (state) => {
-    store.dispatch(changeInstance(state.instance));
-    store.dispatch(filterDatabases(null, state.instance.databases, null));
-    if (state.instance.dataLake && state.instance.dataLake.isDataLake) {
-      store.dispatch(toggleIsDataLake(true));
+  appRegistry.on('instance-refreshed', ({ instance }) => {
+    // First time we are seeing this model, subscribe to status changes
+    if (store.getState().instance === null && instance) {
+      instance.on('change:status', store.onInstanceStatusChange);
```

Comment:
Compass sidebar could've just listened to model change evens instead of global instance refreshed events which would allow us to have more granularranual UI updates and less dependencies on global even emitter but I deliberately decided not to do this in this PR for the reasons mentioned in the PR description